python/cuml/test/test_nearest_neighbors.py

```diff
@@ -657,3 +657,8 @@ def test_haversine_fails_high_dimensions():
                  algorithm='brute')
 
     cunn.fit(data).kneighbors(data)
+
+
+def test_n_jobs_parameter_passthrough():
+    cunn = cuKNN()
+    cunn.set_params(**{"n_jobs": 12})
```

Comment:
Any reason we're avoiding the following?
```suggestion
    cunn.set_params(n_jobs=12)
```

Reply:
Thank you for the comment @wphicks !!

The motivation behind this PR is to allow for [this line of code](https://github.com/scikit-learn-contrib/imbalanced-learn/blob/7e94390cdcd18e6ade70f01e186ebffb3525a8e5/imblearn/over_sampling/_adasyn.py#L127) to not produce an error when integrating a cuKNN() estimator into imbalanced-learn's ADASYN resampler.  I simply copied the problematic call to create this test, but I do prefer your suggestion!